phpstan-baseline.neon

```diff
@@ -1,5 +1,25 @@
 parameters:
 	ignoreErrors:
+		-
+			message: "#^Method loophp\\\\collection\\\\Collection\\:\\:matching\\(\\) return type with generic interface loophp\\\\collection\\\\Contract\\\\Collection does not specify its types\\: TKey, T$#"
+			count: 1
+			path: src/Collection.php
+
+		-
+			message: "#^Return type \\(loophp\\\\collection\\\\Contract\\\\Collection\\) of method loophp\\\\collection\\\\Collection\\:\\:matching\\(\\) should be covariant with return type \\(Doctrine\\\\Common\\\\Collections\\\\Collection\\<mixed, mixed\\>\\) of method Doctrine\\\\Common\\\\Collections\\\\Selectable\\<TKey,T\\>\\:\\:matching\\(\\)$#"
+			count: 1
+			path: src/Collection.php
+
+		-
+			message: "#^Return type \\(loophp\\\\collection\\\\Contract\\\\Collection\\) of method loophp\\\\collection\\\\Collection\\:\\:matching\\(\\) should be covariant with return type \\(loophp\\\\collection\\\\Contract\\\\Collection\\<mixed, mixed\\>\\) of method loophp\\\\collection\\\\Contract\\\\Operation\\\\Matchingable\\<TKey,T\\>\\:\\:matching\\(\\)$#"
+			count: 2
+			path: src/Collection.php
+
+		-
+			message: "#^Return type \\(loophp\\\\collection\\\\Contract\\\\Collection\\<TKey, T\\>\\) of method loophp\\\\collection\\\\Contract\\\\Operation\\\\Matchingable\\:\\:matching\\(\\) should be covariant with return type \\(Doctrine\\\\Common\\\\Collections\\\\Collection\\<mixed, mixed\\>\\) of method Doctrine\\\\Common\\\\Collections\\\\Selectable\\<TKey,T\\>\\:\\:matching\\(\\)$#"
+			count: 1
+			path: src/Contract/Operation/Matchingable.php
```

Comment:
I don't think we should add these to the baseline, something seems to be wrong with the template annotations. Also, a static analysis check in `tests/` would be good - should be fairly simple because the type of the items in the collection doesn't change

Reply:
let me know if you need me to take a stab at these 😅

Reply:
I don't know if we can fix so easily

Reply:
Yeah if you could have a look, I'm unable to get rid of those messages.

Reply:
ok I will look into it today and update you